designsafe/apps/rapid/openTopo_Data/scripts/centerpoint_datagenerator.py

```diff
@@ -103,6 +114,36 @@ def save_aggregated_results(output_filepath, all_features):
         geojson.dump(feature_collection, f, indent=2)
     logging.info(f"Aggregated results saved to: {output_filepath}")
 
+def fetch_otcatalog_api_response(file_path):
+    with open(file_path, 'r') as file1:
+        data = json.load(file1)
+    features = data['features']
+    # Specify the API endpoint URL
+    api_url="https://portal.opentopography.org/API/otCatalog"
+    # Specify default API query parameters
+    include_federated="false"
+    detail="true"
+    for feature in features:
```

Comment:
I follow how for each Data_Catalog_Spatial_Boundaries dataset, we are making a query to otCatalog.  Have we tried just making a query with otCatalog to get all OT datasets (i.e Latitude from -90 to 90 and longitude from -180 to 180). If that was possible, then a single request is made and would simplify things.

Reply:
Yes, Issue resolved when whole world data is fetched from the API call.